### ISSUE TITLE
Update API schema to include opportunity attachments

### ISSUE THREAD

**Issue by chouinar** [HHS/simpler-grants-gov#2466]
### Summary

We currently use the same schema for both the search and GET opportunity endpoint. We likely want a separate schema for non-search (use inheritance so we don't need to redefine everything), as we don't want the attachments on the search schema.

The API response should look something like (trimming out some existing things for brevity):

```json
{
   "opportunity_id": 1234,
   "summary": { .. },
   "attachments": [
       {
         "file_location": "https://...",
         "mime_type": "application/pdf",
         "file_name": "my_NOFO.pdf",
         "file_description": "The full announcement NOFO",
         "file_size_bytes": "10012",
         "created_at": "2024-01-01 12:00:00",
         "updated_at": "2024-01-01 12:00:00",
         "opportunity_attachment_type": "notice_of_funding_opportunity"
       }
   ]
}
```

For futher details on the types of these fields in the API, see https://github.com/HHS/simpler-grants-gov/issues/2123 where we will set this up in the DB. Note that most of these are just strings and integers. The attachment type will be an enum that would have been created in that PR. If you are building this in parallel, make it a string temporarily and we'll switch it later.

A few fields like created_by/updated_by/legacy_folder_id are deliberately not included in the API schema as they're only for internal purposes at the moment.

### Acceptance criteria

- [ ] API schema modified
   - [ ] Examples included for fields that the default isn't helpful (ie. strings default should be changed which isn't helpful)
   - [ ] Descriptions provided for all individual fields to explain what they do
- [ ] The API should fetch the data from the database - NOTE that adjusting the file locations to be something a user can go to will be handled in a follow-up PR
- [ ] Tests added to validate data returned properly from the database for these fields (note you may need to create a factory for the DB table)
